File: ruby/compete/code_golf/dont-panic.rb
STDOUT.sync=1
s=:to_i
def t;gets.split;end
a=t.map(&s)
i=[]
j=[]
a[7].times{k=t.map(&s);i<<k[0];j<<k[1]}
loop{l,m,n=t;m=m.to_i;o=i.index(l.to_i);q=o ?j[o]:a[4];puts (q<m&&n[0]=='L')||(m<q&&n[0]=='R')||q==m ?:WAIT: :BLOCK}

File: python3/compete/code_golf/dont-panic.py
_,_,_,_,a,_,_,b=map(int,input().split())
c,d=[],[]
for i in range(b):e,f=map(int,input().split());c.append(e);d.append(f)
while 1:g,h,i=input().split();g,h=map(int,[g,h]);j=c.index(g)if g in c else-1;k=d[j]if j!=-1else a;print('WAIT'if(k<h and i[0]=='L')or(h<k and i[0]=='R')or k==h else'BLOCK')
